pack/nodejs/helm.ts

```diff
@@ -0,0 +1,715 @@
+    repo?: string;
+
+    // If untar is specified, this flag specifies the name of the directory into which the chart is
+    // expanded (default ".").
+    untardir?: string;
+
+    // Chart repository username.
+    username?: string;
+
+    // Location of your Helm config. Overrides $HELM_HOME (default "/Users/alex/.helm").
+    home?: string;
+
+    // Use development versions, too. Equivalent to version '>0.0.0-0'. If --version is set, this is
+    // ignored.
+    devel?: boolean;
+
+    // Fetch the provenance file, but don't perform verification.
+    prov?: boolean;
+
+    // If set to false, will leave the chart as a tarball after downloading.
+    untar?: boolean;
+
+    // Verify the package against its signature.
+    verify?: boolean;
+}
+
+// Retrieve a package from a package repository, and download it locally.
+//
+// This is useful for fetching packages to inspect, modify, or repackage. It can also be used to
+// perform cryptographic verification of a chart without installing the chart.
+//
+// There are options for unpacking the chart after download. This will create a directory for the
+// chart and uncompress into that directory.
+//
+// If the `verify` option is specified, the requested chart MUST have a provenance file, and MUST
+// pass the verification process. Failure in any part of this will result in an error, and the chart
+// will not be saved locally.
+export function fetch(chart: string, opts?: FetchOpts) {
+    const flags = opts != null ? [
+        // Untar by default.
+        ...opts.untar == false ? [] : [`--untar`],
+
+        ...opts.version != null     ? [`--version ${shell.quote([opts.version])}`]         : [],
+        ...opts.caFile != null      ? [`--ca-file ${shell.quote([opts.caFile])}`]          : [],
+        ...opts.certFile != null    ? [`--cert-file ${shell.quote([opts.certFile])}`]      : [],
+        ...opts.keyFile != null     ? [`--key-file ${shell.quote([opts.keyFile])}`]        : [],
+        ...opts.destination != null ? [`--destination ${shell.quote([opts.destination])}`] : [],
+        ...opts.keyring != null     ? [`--keyring ${shell.quote([opts.keyring])}`]         : [],
+        ...opts.password != null    ? [`--password ${shell.quote([opts.password])}`]       : [],
+        ...opts.repo != null        ? [`--repo ${shell.quote([opts.repo])}`]               : [],
+        ...opts.untardir != null    ? [`--untardir ${shell.quote([opts.untardir])}`]       : [],
+        ...opts.username != null    ? [`--username ${shell.quote([opts.username])}`]       : [],
+        ...opts.home != null        ? [`--home ${shell.quote([opts.home])}`]               : [],
+        ...opts.devel == true       ? [`--devel`]                                          : [],
+        ...opts.prov == true        ? [`--prov`]                                           : [],
+        ...opts.verify == true      ? [`--verify`]                                         : [],
+    ] : [];
```

Comment:
This is pretty strange code :-).  Can this just be a bunch of simple `if`s?

Also:
1. `!==` and `===`
2. `undefined` instead of `null`